### ISSUE TITLE
Enabling SDK Mode?

### ISSUE THREAD

**Issue by zemmyang** [Rexios80/polar#23]
I noticed when I tried to get the gyro stream settings (`polar.requestStreamSettings(identifier, PolarDataType.gyro);`) that the only available settings were 52Hz/2000dps. It looks like changing this is available by enabling SDK mode - does this package implement enabling it?

SDK mode ref: https://github.com/polarofficial/polar-ble-sdk/blob/master/documentation/SdkModeExplained.md

**Comment by Rexios80** [comment]
I tried to add this but am having this issue: https://github.com/polarofficial/polar-ble-sdk/issues/440

I think enabling SDK mode is working though so you can try this if you want
```yaml
polar:
  git:
    url: https://github.com/Rexios80/polar
    ref: 33638d4a43adcd1b488a028f38c85b9a357e145a
```

**Comment by zemmyang** [comment]
Appears to be working on my end (Android). I modified the sample app as follows:

```

  @override
  void initState() {

    // polar
    //     .searchForDevice()
    //     .listen((e) => log('Found device in scan: ${e.deviceId}'));
    polar.batteryLevel.listen((e) => log('Battery: ${e.level}'));
    polar.deviceConnecting.listen((_) => log('Device connecting'));
    polar.deviceConnected.listen((_) => log('Device connected'));
    polar.deviceDisconnected.listen((_) => log('Device disconnected'));

    polar.connectToDevice(identifier);
    // polar.connectToDevice(identifier).then((value) =>
    //     polar.enableSdkMode(identifier)
    // ); // <- doesnt work

    streamWhenReady();

    super.initState();
  }

  Future<void> getSettings() async {
    await polar.enableSdkMode(identifier);

    final settings = await polar.requestStreamSettings(identifier, PolarDataType.ppg);
    log('Settings: ${settings.settings}');
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Polar example app'),
          actions: [
            // PopupMenuButton(
            //   itemBuilder: (context) => RecordingAction.values
            //       .map((e) => PopupMenuItem(value: e, child: Text(e.name)))
            //       .toList(),
            //   onSelected: handleRecordingAction,
            //   child: const IconButton(
            //     icon: Icon(Icons.fiber_manual_record),
            //     disabledColor: Colors.white,
            //     onPressed: null,
            //   ),
            // ),
            IconButton(
              icon: const Icon(Icons.stop),
              onPressed: () {
                log('Disconnecting from device: $identifier');
                polar.disconnectFromDevice(identifier);
              },
            ),
            IconButton(
              icon: const Icon(Icons.play_arrow),
              onPressed: () {
                log('Connecting to device: $identifier');
                polar.connectToDevice(identifier);
                streamWhenReady();
              },
            ),
          ],
        ),
        body: ListView(
          padding: const EdgeInsets.all(10),
          shrinkWrap: true,
          children: logs.reversed.map(Text.new).toList(),
        ),
      ),
    );
  }

  void streamWhenReady() async {

    await polar.sdkFeatureReady.firstWhere(
          (e) =>
      e.identifier == identifier &&
          e.feature == PolarSdkFeature.onlineStreaming,
    );
    final availabletypes = await polar.getAvailableOnlineStreamDataTypes(identifier);

    getSettings();

    debugPrint('available types: $availabletypes');

    //...stream listen, etc.
```

**Comment by Rexios80** [comment]
You have to wait for the sdkMode feature to be ready that's why it doesn't work on connect

**Comment by Rexios80** [comment]
Released in 7.1.0
